fgpyo/util/inspect.py

```diff
@@ -1,16 +1,12 @@
-import functools
```

Comment:
It's too late to propose making `inspect.py` "private" (e.g. `_inspect.py`) but I'm wondering if you think any of the public API is too tempting or dangerous for users to accidentally import and use and rely on. If yes, then we could go and prefix lots of what you added with `_` to hide more things.